src/app/content/Content.tsx

```diff
@@ -22,6 +27,18 @@ export function Content({ term }: ContentProps): JSX.Element {
   const [searchParams] = useSearchParams();
   const { data, loading } = useGetCourse({ term, pid: searchParams.get('pid') || '' });
 
+  const { courses, addCourse, deleteCourse, contains } = useSavedCourses();
+
+  const courseIsSaved = contains(data?.pid!, term);
+
+  const handleBookmarkClick = useCallback(() => {
+    if (!courseIsSaved) {
```

Comment:
If you check if `data` exists in here you can avoid all the checks inside and not have to do the `?` everywhere.